index.js

```diff
@@ -1,124 +1,130 @@
+  first link in a description is the tribe name.
+*/
+const scrape_wiki = async (url, path) => {
+  const response = await axios.get(url)
+  const $ = cheerio.load(response.data)
+
+  const $items = $('h3 ~ table td > table, h2 ~ table td > table')
+  const data = []
+
+  $items.each((i, elem) => {
+    let $elem = $(elem)
+    let $img_a = $elem.find('a.image')
+
+    let file_path = $img_a.attr('href').split('/wiki/File:')[1]
+
+    if (file_path === 'Placeholderflag.png') return
+
+    let title
+    let link = $elem.find('tbody tr:nth-child(2) a:nth-child(1)')
+
+    if (link.length) {
+      title = link.attr('title').split(':').pop()
+    } else {
+      title = $elem.find('tbody tr:nth-child(2)').text().split(',').shift()
+    }
+
+    const item = {
+      img_src: `https://www.mediawiki.org/w/index.php?title=Special:Redirect/file/${file_path}`,
+      title: title,
+      slug: slug(title).replace('flag-', ''),
+      prefix: 'flag-'
+    }
+
+    data.push(item)
+  })
+
+  fs.writeFileSync(`sources/wiki/${path}.json`, JSON.stringify(data, null, 2))
+}
+
+const download_images = async () => {
+  let files = glob.sync('sources/*/*.json')
+  for (file of files) {
+    let data = JSON.parse(fs.readFileSync(file))
+
+    for (item of data) {
+      const response = await axios.get(item.img_src, {
+        responseType: 'stream'
+      })
+      const ext = path
+        .extname(url.parse(response.request.res.responseUrl).pathname)
+        .toLowerCase()
+      const out_file = `./unprocessed/${item.prefix || ''}${item.slug}${
+        ext || ''
+      }`
+      const stream = fs.createWriteStream(out_file)
+      response.data.pipe(stream)
+
+      stream.on('finish', async () => {
+        if (ext === '.svg') {
+          const cleaned = (await svgo.optimize(fs.readFileSync(out_file))).data
+          fs.writeFileSync(out_file, cleaned)
+        }
+      })
+    }
+  }
+}
+
+const resize_images = async () => {
+  glob('unprocessed/*', {}, async (er, files) => {
+    for (file of files) {
+      const ext = path.extname(file)
+      const basename = path.basename(file).replace(ext, '')
+      let quality = 1.0
+
+      try {
+        let blob = await sharp(file).toBuffer()
+
+        do {
+          console.log(basename, ext, file, quality)
+          blob = await sharp(blob)
+            .resize(MAX_WIDTH, MAX_HEIGHT, {
+              fit: sharp.fit.inside,
+              withoutEnlargement: true
+            })
+            .sharpen()
+            .png()
+            .toBuffer()
+
+          quality -= 0.05
+        } while (quality > 0.4 && blob.size > MAX_FILE_SIZE)
+
+        sharp(blob).toFile(`dist/${basename}.png`)
+      } catch (e) {
+        console.log(`Error processing ${basename}: ${e}`)
+      }
+    }
+  })
+}
+
+const main = async () => {
+  await scrape_wiki(US_URL, 'us')
+  await scrape_wiki(CA_URL, 'ca')
+
+  await download_images()
+  await resize_images()
+}
+
+main()
```

Comment:
Are there any major programmatic changes here or just prettier? Hard to track with the prettification.

Reply:
No programmatic changes, only prettier

Reply:
It's easier if you add `?w=1` at the end of the commit url

example: https://github.com/nativesintech/indigemoji/pull/7/commits/58046b5db7bf5dd1413833df4bf6e9f6ea2121f7?w=1

It ignores white-space changes and such and is way easier to read, re: https://github.blog/2011-10-21-github-secrets/#whitespace

Reply:
AH! I see. That helps. I always forget about that.